src/lib/bootloader/mbr_update.rb

```diff
@@ -166,21 +178,21 @@ def bios_id_for(device)
 
     # List of partition for disk that can be used for setting boot flag
     def activatable_partitions(disk)
-      partitions = target_map.fetch(disk, {}).fetch("partitions", [])
+      return [] unless disk
+
       # do not select swap and do not select BIOS grub partition
       # as it clear its special flags (bnc#894040)
-      partitions.select do |p|
-        p["used_fs"] != :swap && p["fsid"] != Yast::Partitions.fsid_bios_grub
+      devicegraph.disks.with(name: disk.name).partitions.reject do |part|
+        [Storage::ID_SWAP, Storage::ID_GPT_BIOS].include?(part.id)
       end
```

Comment:
this is not how it works before. Previously it was more reboust if disk is partitionaless or if disk is not passed. what happen here if disk is not found?